### PR TITLE
daml-lf: Move archive protofile to a more usual place.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
move proto definitions from archive to a more classical place.

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Archive.hs [hunk 1]
```diff
@@ -14,7 +14,7 @@ module DA.Daml.LF.Proto3.Archive
 
 import           Control.Lens             (over, _Left)
 import qualified "cryptonite" Crypto.Hash as Crypto
-import qualified Da.DamlLf                as ProtoLF
+import qualified Com.Digitalasset.DamlLf.DamlLf as ProtoLF
```
Comment (cocreature) [inline compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Archive.hs]: Now our Haskell looks like Java :cry:

File: daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf/daml_lf.proto [hunk 2]
```diff
@@ -5,10 +5,10 @@ syntax = "proto3";
 package daml_lf;
 
 option java_package = "com.digitalasset.daml_lf";
-option csharp_namespace = "Com.DigitalAsset.Daml_lf.DamlLf";
+option csharp_namespace = "Com.DigitalAsset.Daml_lf.DamlLfDev";
```
Comment (cocreature) [inline daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf/daml_lf.proto]: This seems weird, why do we only change this for `csharp` but not for `java`?

If that is really intended it needs an entry in the release notes.

File: compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Decode.hs [hunk 3]
```diff
@@ -6,7 +6,7 @@ module DA.Daml.LF.Proto3.Decode
   , decodePayload
   ) where
 
-import Da.DamlLf (ArchivePayload(..), ArchivePayloadSum(..))
+import Com.Digitalasset.DamlLf.DamlLf (ArchivePayload(..), ArchivePayloadSum(..))
```
Comment (cocreature) [inline compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Decode.hs]: I guess the reason why this changes for Haskell but not for Java/Scala is due to the `java_package` annotation?

File: daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf/daml_lf.proto (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf/daml_lf.proto]: My mistake. This is for the next PR.

File: compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Decode.hs (hunk 3 above)
Comment (remyhaemmerle-da) [reply compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Decode.hs]: Indeed. As far I can tell haskell proto3-suite do not provide the equivalent.

File: daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf/daml_lf.proto (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf/daml_lf.proto]: fixed
